crates/infra/cli/src/commands/setup/cargo/mod.rs

```diff
@@ -2,6 +2,61 @@ use anyhow::Result;
 use infra_utils::{commands::Command, github::GitHub};
 
 pub fn setup_cargo() -> Result<()> {
+    // The bootstrap bash script defined in '$REPO_ROOT/scripts/_common.sh'
+    // installs the 'minimal' profile of the '$RUST_STABLE_VERSION' toolchain.
+    // This includes the following components:
+    //
+    // - 'cargo'
+    // - 'rust-std'
+    // - 'rustc'
+    //
+    // Which are enough to run infra scripts.
+    // But we need these additional optional components for local development:
+
+    rustup_add_components(env!("RUST_STABLE_VERSION"), ["clippy"])?;
+
+    if !GitHub::is_running_in_ci() {
+        rustup_add_components(
+            env!("RUST_STABLE_VERSION"),
+            ["rust-analyzer", "rust-docs", "rust-src"],
+        )?;
+    }
+
+    // Additionally, we also need 'rustfmt nightly', as we use experimental options.
+    // So let's install the '$RUST_NIGHTLY_VERSION' toolchain along with the 'rustfmt' component.
+
+    rustup_install_toolchain(env!("RUST_NIGHTLY_VERSION"))?;
+
+    rustup_add_components(env!("RUST_NIGHTLY_VERSION"), ["rustfmt"])?;
+
+    // Make sure we have the latest dependencies:
+
```

Comment:
Is there a reason for so many empty lines here? I'm fine with separating related blocks of code with a single newline in general, but adding a newline between a comment and the line/logic that the comment refers to seems extraneous/confusing here.

Reply:
Just for consistency with the code above. There is an empty line between each block of comments, and the block of code after it.
No strong feelings about it.

Reply:
IMO it adds a lot of noise and I'd expect the newline to separate blocks that are less related (i.e. we group related code but separate that which is not) but in this case we explicitly separate the comment and the relevant lines that the comment refers to.

I'd expect:
```
// Do the thing.
do_thing();

the_thing_with_some_extra();

// Now another
another();
```

and not:
```
// Do the thing.

do_thing();

the_thing_with_some_extra();

// Now another

another();
```

which, by uniformly adding space between everything, effectively negates any grouping and makes it seem like an extraneous line spacing rather than grouping related code.

If that's okay, could we structure it at least a little bit and not include a line between everything? :sweat_smile: :pray: Definitely nitpicking but seems a bit unorganized/harder to focus on the important bits, even if it is conceptually trivial.

Reply:
Sure.